src/core/device.c

```diff
@@ -659,10 +659,8 @@ static void device_shutdown(Manager *m) {
 
         m->udev_event_source = sd_event_source_unref(m->udev_event_source);
 
-        if (m->udev_monitor) {
-                udev_monitor_unref(m->udev_monitor);
-                m->udev_monitor = NULL;
-        }
+        if (m->udev_monitor)
+                m->udev_monitor = udev_monitor_unref(m->udev_monitor);
```

Comment:
this can be shortended even further: udev_monitor_unref() is a NOP when NULL is passed in, hence the if check can be dropped too

Reply:
Oh, indeed. I may update this later.

Reply:
#8661